src/pages/register-a-beacon/primary-beacon-use.tsx

```diff
@@ -16,30 +20,57 @@ import {
   RadioListItemConditional,
   RadioListItemHint,
 } from "../../components/RadioList";
-import { withCookieRedirect } from "../../lib/middleware";
+import { CacheEntry } from "../../lib/formCache";
+import { FormValidator } from "../../lib/formValidator";
+import { handlePageRequest } from "../../lib/handlePageRequest";
 import { MaritimePleasureVessel } from "../../lib/types";
+import { ensureFormDataHasKeys } from "../../lib/utils";
+
+interface PrimaryBeaconUseProps {
+  formData: CacheEntry;
+  needsValidation: boolean;
+}
 
 interface BeaconUseFormProps {
-  heading: string;
+  formData: CacheEntry;
+  checkedValue: string | null;
+  showErrors: boolean;
+  errorMessages: string[];
 }
 
-const PrimaryBeaconUse: FunctionComponent = (): JSX.Element => {
-  const pageHeading =
-    "What type of maritime pleasure vessel will you mostly use this beacon on?";
+const PrimaryBeaconUse: FunctionComponent<PrimaryBeaconUseProps> = ({
+  formData,
+  needsValidation = false,
+}: PrimaryBeaconUseProps): JSX.Element => {
+  formData = ensureFormDataHasKeys(
+    formData,
+    "maritimePleasureVesselUse",
+    "otherPleasureVesselText"
+  );
 
-  // TODO: Use form validation to set this
-  const pageHasErrors = false;
+  const errors = FormValidator.errorSummary(formData);
 
   return (
     <Layout
+      title={
+        "What type of maritime pleasure vessel will you mostly use this beacon on?"
+      }
       navigation={<BackButton href="/register-a-beacon/beacon-information" />}
-      title={pageHeading}
-      pageHasErrors={pageHasErrors}
+      pageHasErrors={errors.length > 0 && needsValidation}
     >
       <Grid
         mainContent={
           <>
-            <BeaconUseForm heading={pageHeading} />
+            {needsValidation && <FormErrorSummary errors={errors} />}
```

Comment:
Just a thought, as we're doing this on most pages is it worth exposing an input param to the `<FormErrorSummary />` component that handles the conditional rendering here?

Reply:
I like this!  Actioned 👍🏻